### DISCUSSION TITLE
API call to get all interfaces for a specific virtual machine

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#8884] [Q&A] [answered]
Hi

I am looking to get a list of all interface that are on a specific virtual machine using pynetbox. 

I thought it was something like this:

```
nb = pynetbox.api(url='https://netbox-server/', token='123456789')
ns_vs_int=nb.virtualization.interfaces.filter(virtual_machine='test-server')
print (nb_vs_int)
<pynetbox.core.response.RecordSet object at 0x7f5c2d202828>
```

Trying 
`ns_vs_int=nb.virtualization.interfaces.filter(virtual_machine_id=234)`
doesn't make a difference.

What am I doing wrong?

**Comment by markkuleinio** [comment, marked as answer]
Hi, maybe nothing, for printing you have to make it a list:

```
my_list = list(ns_vs_int)
print(my_list)
```
or just iterate over the results:
```
for i in ns_vs_int:
    print(i)
```

RecordSet is a Python generator so it is not a list.

**Reply by nomad-cyanide** [reply]
Thank you! I'll go take a nap now, since I must be tired when I couldn't see that. :-/
